torchvision/transforms/_core.py

```diff
@@ -23,17 +24,40 @@ def query_sample(sample: Any, fn: Callable[[Any], Optional[T]]) -> Iterator[T]:
             yield result
 
 
+class TransformDispatch:
```

Comment:
IMO something like `JointTransform` implies its use case better.

```suggestion
class JointTransform:
```